src/core/auth/qgsauthconfigurationstorage.h

```diff
@@ -141,82 +152,93 @@ class CORE_EXPORT QgsAuthConfigurationStorage: public QObject
      * by one.
      */
-    virtual QgsAuthMethodConfigsMap authMethodConfigsWithPayload( ) const = 0;
+    virtual QgsAuthMethodConfigsMap authMethodConfigsWithPayload( ) const SIP_THROW( QgsNotSupportedException ) = 0;
 
     /**
      * Load an authentication configuration from the database.
      * \param id Configuration id.
      * \param payload (possibly encrypted) payload.
      * \param full If TRUE, the full configuration is loaded and the (possibly encrypted) payload is populated, otherwise only the configuration metadata is loaded.
      * \returns Authentication configuration metadata.
+     * \throws QgsNotSupportedException if the operation is not supported by the storage.
      */
-    virtual QgsAuthMethodConfig loadMethodConfig( const QString &id, QString &payload SIP_OUT, bool full = false ) const = 0;
+    virtual QgsAuthMethodConfig loadMethodConfig( const QString &id, QString &payload SIP_OUT, bool full = false ) const SIP_THROW( QgsNotSupportedException ) = 0;
 
     /**
      * Store an authentication config in the database.
      * \param config Authentication configuration.
      * \param payload payload to store (possibly encrypted).
      * \returns Whether operation succeeded
+     * \throws QgsNotSupportedException if the operation is not supported by the storage.
      */
-    virtual bool storeMethodConfig( const QgsAuthMethodConfig &config, const QString &payload ) = 0;
+    virtual bool storeMethodConfig( const QgsAuthMethodConfig &config, const QString &payload ) SIP_THROW( QgsNotSupportedException ) = 0;
 
     /**
      * Removes the authentication configuration with the specified \a id.
      * \returns TRUE if the configuration was removed, FALSE otherwise.
+     * \throws QgsNotSupportedException if the operation is not supported by the storage.
      */
-    virtual bool removeMethodConfig( const QString &id ) = 0;
+    virtual bool removeMethodConfig( const QString &id ) SIP_THROW( QgsNotSupportedException ) = 0;
 
     /**
      *  Check if an authentication configuration exists in the storage.
      *  \param id Configuration id.
      *  \returns TRUE if the configuration exists, FALSE otherwise.
+     *  \throws QgsNotSupportedException if the operation is not supported by the storage.
      */
-    virtual bool methodConfigExists( const QString &id ) const = 0;
+    virtual bool methodConfigExists( const QString &id ) const SIP_THROW( QgsNotSupportedException ) = 0;
 
     /**
      * Store an authentication setting in the storage.
      * \param key Setting key.
      * \param value Setting value.
      * \returns Whether operation succeeded
+     * \throws QgsNotSupportedException if the operation is not supported by the storage.
      */
-    virtual bool storeAuthSetting( const QString &key, const QString &value ) = 0;
+    virtual bool storeAuthSetting( const QString &key, const QString &value ) SIP_THROW( QgsNotSupportedException ) = 0;
```

Comment:
The corresponding methods in QgsAuthManager will also need the exception documentation and SIP_THROW

Reply:
As discussed, this is not necessary because the error conditions are pre-checked and the exceptions would never be thrown and bubble up to the auth manager. In other words: it cannot happen.